Userland/Libraries/LibJS/Runtime/PropertyName.h

```diff
@@ -113,6 +113,19 @@ class PropertyName {
     bool try_coerce_into_number()
     {
         VERIFY(m_string_may_be_number);
+        if (m_string.is_empty()) {
```

Comment:
Not sure if we want to do this here or add other `TrimWhitespace` like flags to `to_int`.
Might get very cluttered with too many flags / javascript specific